ibtk/include/ibtk/FEProjector.h

```diff
@@ -0,0 +1,123 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2011 - 2019 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+#ifndef included_IBTK_FEProjector
+#define included_IBTK_FEProjector
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <libmesh/equation_systems.h>
+#include <libmesh/petsc_linear_solver.h>
+#include <libmesh/petsc_matrix.h>
+#include <libmesh/petsc_vector.h>
+
+#include <map>
+#include <string>
+
+namespace IBTK
+{
+class FEData;
+} // namespace IBTK
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBTK
+{
+/*!
+ * \brief Class FEProjector coordinates data structures for projecting
+ * fields in FE models.
+ */
+class FEProjector
+{
+public:
```

Comment:
We also need a `clearCached` function here that resets (at least) any data dependent on the Eulerian partitioning.

We don't have proper separation of concerns right now regarding regridding libMesh data - I think this is done in IBFEMethod. We really need some clear distinction here between 'update the IB data' and 'update libMesh's data that depends on its own parallel partitioning'.

Reply:
Perhaps we should change the method name to something other than `clearCached` --- maybe -- this is kind of lengthy, but something like `clearCartGridDependentData()` or `clearCartPartitionDependentData()`?

Reply:
I propose `clearPatchHierarchyDependentData`.